File: api/models/DataProcessor30ListingData.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.EntityFrameworkCore;

namespace api.Models
{
    public class DataProcessor30ListingData
    {
        public int Id { get; set; }
    
        public string Name { get; set; } = string.Empty;
        
        public string Solution { get; set; } = string.Empty;

        public byte Archived { get; set; } = 0;


        public DateTime CreationTime {get; set; } = DateTime.Now;
        public DateTime UpdateTime {get; set; } = DateTime.Now;

        
        public DataController DataController { get; set; } = new DataController();
        public DataProcessor DataProcessor { get; set; } = new DataProcessor();
        public DataProcessorRepresentative DataProcessorRepresentative { get; set; } = new DataProcessorRepresentative();
        public DataControllerRepresentative DataControllerRepresentative { get; set; } = new DataControllerRepresentative();
        public DataSecurityAdvisor DataSecurityAdvisor { get; set; } = new DataSecurityAdvisor();

        public DataCategories DataCategories { get; set; } = new DataCategories();
        public DataTransfer DataTransfer { get; set; } = new DataTransfer();
        public DataSecurity DataSecurity {get; set; } = new DataSecurity();

        public List<DataEdit> DataEdits { get; set; } = new List<DataEdit>();
        public List<DataSubProcessor> DataSubProcessors { get; set; } = new List<DataSubProcessor>();
        
    }
}

File: frontend/src/types/DataProcessor30ListingData.ts
export type DataProcessor30ListingDataDto = Omit<
   DataProcessor30ListingData,
   "id" | "creationTime" | "updateTime"
>;

type DataProcessor30ListingData = {
   solution: string;
   id: number;
   name: string;
   creationTime: Date;
   updateTime: Date;
   dataController: DataController;
   dataProcessor: DataProcessor;
   dataControllerRepresentative: DataControllerRepresentative;
   dataProcessorRepresentative: DataProcessorRepresentative;
   dataCategories: DataCategories;
   dataSecurity: DataSecurity;
   dataSubProcessors: SubProcessor[];
   dataEdits: DataEdits[];
};

export type DataContactInfo = {
   name: string;
   address: string;
   phoneNo: string;
   mail: string;
};

export type DataController = DataContactInfo & {
   cvr: string;
};

export type DataProcessor = DataContactInfo & {
   cvr: string;
};

export type DataControllerRepresentative = DataContactInfo & {
   role: string;
};

export type DataProcessorRepresentative = DataContactInfo & {
   role: string;
};

export type DataCategories = {
   categoryList: string;
};

export type DataSecurity = {
   securityMeasures: string;
};

export type DataTransfer = {
   transferInformation: string;
};

export type DataEdits = {
   id: number;
   editType: string;
   editTime: Date;
   comment: string;
   dataProcessor30ListingDataId: number;
   fieldsEdited: string;
};

export type SubProcessor = {
   name: string;
   cvr: string;
   address: string;
   treatment: string;
   directSubProcessor: boolean;
   transferReason: string;

   // Utility variable
   parentCompany: string;
};

export default DataProcessor30ListingData;
